### PR TITLE
kvclient(ticdc): add worker busy monitor

### LINKED ISSUE #10388: record worker busy ratio in kvclient
To diagnose whether there is a bottleneck in the kvclient, we need to monitor the goroutines' busy ratio.

### PR DESCRIPTION
**Pull request by CharlesCheung96**
### What problem does this PR solve?

Issue Number: close #10388

#### Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

### Release note 

```release-note
Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with `None`.
```

### CODE REVIEW

File: cdc/kv/metrics.go [hunk 1]
```diff
@@ -143,6 +143,21 @@ var (
 		},
 		// actions: wait, run.
 		[]string{"namespace", "changefeed"})
+
+	workerBusyRatio = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "ticdc",
+			Subsystem: "kvclient",
+			Name:      "worker_busy_ratio",
```
Comment (asddongmen) [inline cdc/kv/metrics.go:151]:
```suggestion
			Name:      "region_worker_busy_ratio",
```

File: cdc/kv/metrics.go [hunk 2]
```diff
@@ -143,6 +143,21 @@ var (
...
+			Help:      "Busy ratio (X ms in 1s) for region worker.",
```
Comment (asddongmen) [inline cdc/kv/metrics.go:152]: Please add more detail about what can we learn from this metric. It will be very helpful.

**Review by asddongmen — APPROVED** [review]

File: cdc/kv/client.go [hunk 3]
```diff
@@ -393,16 +394,19 @@ func newEventFeedSession(
 		tableID:    client.tableID,
 		tableName:  client.tableName,
 
-		totalSpan:         totalSpan,
-		eventCh:           eventCh,
-		rangeLock:         rangeLock,
-		lockResolver:      lockResolver,
-		id:                idStr,
```
Comment (hicqu) [inline cdc/kv/client.go:400]: Seems id is missed.

File: cdc/kv/client.go [hunk 4]
```diff
@@ -1089,10 +1093,32 @@ func (s *eventFeedSession) receiveFromStream(
 	})
 
 	receiveEvents := func() error {
+		metricWorkerBusyRatio := workerBusyRatio.WithLabelValues(
+			s.changefeed.Namespace, s.changefeed.ID, strconv.FormatInt(s.tableID, 10), "event-receiver", addr)
```
Comment (hicqu) [inline cdc/kv/client.go:1097]: It should be moved out from the function.

File: cdc/kv/client.go [hunk 5]
```diff
@@ -1089,10 +1093,32 @@ func (s *eventFeedSession) receiveFromStream(
...
+		metricsTicker := time.NewTicker(time.Second * 10)
+		defer metricsTicker.Stop()
+		var receiveTime time.Duration
+		startToWork := time.Now()
+
 		maxCommitTs := model.Ts(0)
 		for {
+			startToReceive := time.Now()
```
Comment (hicqu) [inline cdc/kv/client.go:1105]: Seems it's only used if enableTableMonitor is true.

File: cdc/kv/metrics.go [hunk 6]
```diff
@@ -143,6 +143,21 @@ var (
 		},
 		// actions: wait, run.
 		[]string{"namespace", "changefeed"})
+
+	workerBusyRatio = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Namespace: "ticdc",
+			Subsystem: "kvclient",
+			Name:      "region_worker_busy_ratio",
```
Comment (hicqu) [inline cdc/kv/metrics.go:151]: Should be renamed to kvclient_receiver_busy_ratio.

Comment (CharlesCheung96) [reply cdc/kv/metrics.go:151]: The monitor also records information about the eventHandler worker.

File: cdc/kv/client.go [hunk 7]
```diff
@@ -1089,10 +1097,36 @@ func (s *eventFeedSession) receiveFromStream(
...
 			cevent, err := stream.Recv()
 
+			if s.enableTableMonitor {
+				receiveTime += time.Since(startToReceive)
+				select {
+				case <-metricsTicker.C:
```
Comment (hicqu) [inline cdc/kv/client.go:1114]: A ticker is not necessary. We only need a variable.

Comment (hicqu) [reply cdc/kv/client.go:1114]: And `10s` or `20s` seems too long. How about use `2s`?

**Review by hicqu — APPROVED** [review]

**Pull request merged**